curriculum/challenges/english/25-front-end-development/lab-polygon-area-calculator/5e444147903586ffb414c94f.md

```diff
@@ -0,0 +1,881 @@
+---
+id: 5e444147903586ffb414c94f
+title: Build a Polygon Area Calculator
+challengeType: 27
+dashedName: build-a-polygon-area-calculator
+---
+
+# --description--
+
+In this project, you will use object-oriented programming to create a `Rectangle` class and a `Square` class. The `Square` class should be a subclass of `Rectangle` and inherit its methods and attributes.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should create a `Rectangle` class.
+2. When a `Rectangle` object is created, it should be initialized with `width` and `height` attributes. The class should also contain the following methods:
+
+    - `set_width`: Sets the width of the rectangle.
+    - `set_height`: Sets the height of the rectangle.
+    - `get_area`: Returns area (`width * height`).
+    - `get_perimeter`: Returns perimeter (`2 * width + 2 * height`).
+    - `get_diagonal`: Returns diagonal (`(width ** 2 + height ** 2) ** .5`).
+    - `get_picture`: Returns a string that represents the shape using lines of `*`. The number of lines should be equal to the height and the number of '\*' in each line should be equal to the width. There should be a new line (`\n`) at the end of each line. If the width or height is larger than `50`, this should return the string: `'Too big for picture.'`.
```

Comment:
it looks like this should be in backticks too. otherwise it looks like this rendered on the screen.

<img width="447" alt="Screenshot 2025-05-06 at 5 01 23 PM" src="https://github.com/user-attachments/assets/3a1fcf3c-0507-4bcc-a836-4bbd6c4e0c9b" />

```suggestion
    - `get_picture`: Returns a string that represents the shape using lines of `*`. The number of lines should be equal to the height and the number of `*` in each line should be equal to the width. There should be a new line (`\n`) at the end of each line. If the width or height is larger than `50`, this should return the string: `'Too big for picture.'`.
```